pkg/ring/lifecycler.go

```diff
@@ -447,6 +455,11 @@ heartbeatLoop:
 func (i *Lifecycler) initRing(ctx context.Context) error {
 	var ringDesc *Desc
 
+	var tokensFromFile Tokens
+	if err := tokensFromFile.LoadFromFile(i.cfg.TokensFilePath); err != nil {
```

Comment:
Currently this code handles `TokensFilePath` being empty by trying to open the file with empty filename, and (hopefully) discovering that it doesn't exist. It is a strange pattern. It would be clearer to just check `TokensFilePath` for emptiness and skip loading of tokens in such case. If tokens file doesn't exist, it would be nice to log that information as a warning, but continue with no tokens.